packages/angular/ssr/src/routes/route-tree.ts

```diff
@@ -128,29 +114,28 @@ export class RouteTree<AdditionalMetadata extends Record<string, unknown> = {}>
   insert(route: string, metadata: RouteTreeNodeMetadataWithoutRoute & AdditionalMetadata): void {
     let node = this.root;
     const segments = this.getPathSegments(route);
-    const normalizedSegments: string[] = [];
 
-    for (const segment of segments) {
-      // Replace parameterized segments (e.g., :id) with a wildcard (*) for matching
-      const normalizedSegment = segment[0] === ':' ? '*' : segment;
-      let childNode = node.children.get(normalizedSegment);
+    for (let index = 0; index < segments.length; index++) {
+      let segment = segments[index];
+      if (segment[0] === ':') {
+        // Replace parameterized segments (e.g., :id) with a wildcard (*) for matching
+        segments[index] = segment = '*';
```

Comment:
Suggestion: Personally I find this kind of mutation hard to follow and the double assignment makes it particularly tricky to notice that `segment` is updated too.

Could we do this transform separately first? It would also allow us to drop the manual iteration with `index`.

```typescript
const segs = segments.map((segment) => segment.startsWith(':') ? '*' : segment);
for (const segment of segs) {
  // ...
}
```

(Should use a better name than `segs`, but hopefully that gets the point across.)